be/src/formats/parquet/column_chunk_reader.cpp

```diff
@@ -140,6 +140,9 @@ Status ColumnChunkReader::_read_and_decompress_page_data(uint32_t compressed_siz
     Slice read_data;
     auto ret = _page_reader->peek(read_size);
     if (ret.ok()) {
+        DCHECK_EQ(ret.value().size, read_size);
```

Comment:
Are you sure this will always be true under your implementations?

Reply:
yes. if peekable, then always true.

Reply:
Ok, please note this is implementation-defined, there is no guarantee that all future `peek()` implementations will behave in the same way.

Reply:
But this is not defined by the interface, I think we should coding to the interface but not the implementation. IMO, we should either change the usage or the interface definition.

Reply:
> But this is not defined by the interface, I think we should coding to the interface but not the implementation. IMO, we should either change the usage or the interface definition.

I prefer to update the usage.

Reply:
OK, I'll update it.